esphome/components/sml/sml_parser.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include <string>
+#include <vector>
+#include "constants.h"
+
+namespace esphome {
+namespace sml {
+
+using bytes = std::vector<uint8_t>;
+
+class SmlBase {
+ public:
+  uint8_t type;
+  uint8_t length;
+  SmlBase(const bytes &buffer, unsigned int &pos);
+  bool is_list();
+  bool has_extended_length();
+
+ protected:
+  const bytes &buffer_;
+  const unsigned int startpos_;
```

Comment:
```suggestion
  const size_t startpos_;
```

To keep the compiler from complaining when we support 64bit devices

Reply:
Done